tests/sampler/test_SamplerSpacing.py

```diff
@@ -0,0 +1,61 @@
+import pandas
+from map2loop.sampler import SamplerSpacing
+from beartype.roar import BeartypeCallHintParamViolation
+import pytest
+import shapely
+import geopandas
+
+# add test for SamplerSpacing specifically
+
+
+@pytest.fixture
+def sampler_spacing():
+    return SamplerSpacing(spacing=1.0)
+
+
+@pytest.fixture
+def correct_geodata():
+    data = {
+        'geometry': [
+            shapely.LineString([(0, 0), (1, 1), (2, 2)]),
+            shapely.Polygon([(0, 0), (1, 1), (1, 0), (0, 0)]),
+            shapely.MultiLineString(
+                [shapely.LineString([(0, 0), (1, 1)]), shapely.LineString([(2, 2), (3, 3)])]
+            ),
+        ],
+        'ID': ['1', '2', '3'],
+    }
+    return geopandas.GeoDataFrame(data, geometry='geometry')
+
+
+@pytest.fixture
+def incorrect_geodata():
+    data = {'geometry': [shapely.Point(0, 0), "Not a geometry"], 'ID': ['1', '2']}
+    return pandas.DataFrame(data)
+
+
+# test if correct outputs are generated from the right input
+def test_sample_function_correct_data(sampler_spacing, correct_geodata):
```

Comment:
Can the test check that the extracted points are the correct ones not just the correct format? Calculate the distance between the points and expected points and make sure this sums to 0

Reply:
@lachlangrose thanks for reviewing. Can you please confirm if the changes to the test in https://github.com/Loop3D/map2loop/pull/80/commits/37125f480afa2da88b4bef90db985110f6d89307 suffice?